open_mastr/xml_download/utils_write_to_database.py

```diff
@@ -212,18 +211,18 @@ def add_table_to_database(
                     )
                     continueloop = False
         except sqlalchemy.exc.OperationalError as err:
-            add_missing_column_to_table(err, engine, xml_tablename)
+            add_missing_column_to_table(err, engine, xml_tablename, df)
```

Comment:
In my opinion it would make more sense to move the add_missing_column_to_table up to do it before writing data to the table.
Something like:
```
        add_missing_column_to_table(engine, xml_tablename, required_columns)
        try:
            with engine.connect() as con:
                with con.begin():
                    df.to_sql(
                        sql_tablename,
                        con=con,
                        index=False,
                        if_exists=if_exists,
                        dtype=dtypes_for_writing_sql,
                    )
        except sqlalchemy.exc.IntegrityError:
            # error resulting from Unique constraint failed
            df = write_single_entries_until_not_unique_comes_up(
                df=df, xml_tablename=xml_tablename, engine=engine
               
```